### PR TITLE
feat(radio): modify Sensor Edit screens to show Ratio + Percentage

### LINKED ISSUE #4648: feat(telemetry): Edit Sensors: display the percentage (%) equivalent to ratio when scaling a sensor value
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

Not a bug, just enhancement.

The original ratio was design to correct for ADC Voltages to display the correct voltage.

Some people who knows how to compute a voltage divider knows what number to type if they know the electrical schematics of the RX voltage input.

For a big majority, they really don't know where to start unless is documented in the RX manual (or somewhere else).  They just start changing the number until they get the voltage they see in a multimeter.

A "ratio" of 0 (display as "-") mean bypass the scaling logic, so it give the raw value coming from the telemetry screen.
A "ratio" of 25.5 will give you the original sensor value (or 100%).   
Could be helpful to display the percentage on the side of the ratio, so if a user needs to scale up, they know to go to a value higher than 100%,

Some people are using "ratio" for sensors who are not voltages. For example, If you have a Current (A) reading, you could compute a new sensor for consumption by adding this current periodically. Then adjust "ratio" +/- until you get a consumption closer to reality at the end of the flight.

### Describe the solution you'd like

Display the equivalent percentage on the side of the "ratio" field.

Example:
image

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by frankiearzu**
Fixes #4648
**Works in main as well as 2.9 branch.**
Changes are isolated in the display portion of the UI

Summary of changes:
Change the Telemetry Sensor Edit to show the equivalent percentage.
Works on color and B&W radios.

image
image

image
image

### CODE REVIEW

**Comment by pfeerick** [comment]
We are past feature freeze, so this will be 2.11

**Comment by frankiearzu** [comment]
Firmware for Radiomaster TX16S if you want to try it (2.9.4 + this change)
[tx16s-edgetx-v2.9.4-Tel-Ratio-Percent.bin.zip](https://github.com/EdgeTX/edgetx/files/14305937/tx16s-edgetx-v2.9.4-Tel-Ratio-Percent.bin.zip)

**Comment by philmoz** [comment]
My take on this would be to hide the ratio value/percentage altogether and instead show the adjusted sensor value as you edit the field.

This is how the battery calibration field works on the hardware page - you don't see the calibration factor, you just see the adjusted battery voltage.

This remove any confusion over the meaning and gives you real time feedback of what the change is doing to the sensor value.

**Comment by 3djc** [comment]
I like the idea

**Comment by jtaylor2** [comment]
> My take on this would be to hide the ratio value/percentage altogether and instead show the adjusted sensor value as you edit the field.
> 

Then how would you enter the ratio without the sensor present?  If I build a V1 or V2 voltage divider I know what ratio value to enter (it's easily calculated) and for v1 and v2 sensors the values for ratio have been documented in the instructions for years.

I guess I don't understand the current confusion (if it even exists) because it's been the way it is for over a decade and is well documented in many places.  Lets not dumb it down too much and break all the existing documentation. 

https://www.frsky-rc.com/wp-content/uploads/Downloads/Amanual/FBVS01%20Manual.pdf
https://doc.open-tx.org/manual-for-opentx-2-2/software-overview/model_menus/telemetry
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#FLYSKY-AFHDS2A-RX---56

**Comment by frankiearzu** [comment]
Some people like to see the ratio..On many protocols/rx (like FrSky) is documented what value to put in the ratio field, and adjust the offset.   In the beginning I only wanted to do percent but a lot of people commented in Discord about the documented ratios, as well if you know the voltage divider circuit, you can calculate the proper ratio.Also some other people (and me) are using ratio for non-voltage, for example, current consumption if you have realtime current only (like spektrum ESCs). You create a calculated-consumption sensor using the raw current, but you adjust the original realtime current sensor to get close to the real mAh usage on the battery, that is a bit below.. so the realtime current has a ratio of 24.6 (96%), and that gives about the real mAh consumption. In this case, I know that I want a bit below 100%, so precentage makes more sense.Recently someone also use it for angles.. the angle came x10 so he wanted to /10.. so 10% that is ratio of about 2.5. (The computed multiply only works with two sensors, not constants).There is no single answer.. people are using ratio in very creative ways.. not only for voltage.That’s why I think keeping ratio is important, but having a way to see the equivalent in present age is very useful too.On Mar 16, 2024, at 4:02 PM, 3djc ***@***.***> wrote:﻿
I like the idea

**Comment by 3djc** [comment]
I could agree for percentages, but not for meaningless numbers just because old documentation exists.

**Comment by jtaylor2** [comment]
The numbers are hardly meaningless.  A 4 :1 ratio voltage sensor on a 3.3 Volt ADC input  is 4 X 3.3-13.2 for the ratio.  I don't find that meaningless.  It lets me know the max voltage I can apply to that sensor. I find that a lot more useful than the weight which is meaningless to me, But why not have it both ways?  I really don't see any reason to dumb it down.

**Comment by pfeerick** [comment]
@philmoz Do the changes I just made make sense?

bw212 currently looks like this (only minimal change - % suffix trails number properly):
image

bw128 now ditches the `.0` once you hit 1000% and converts ratio to try and fit bigger and bigger numbers... can only do such much though once it gets ridiculously high...
image
image
image

**Comment by philmoz** [comment]
You could move both numbers a bit to the left when the ratio gets higher.

**Comment by frankiearzu** [comment]
Thank you for the nice formatting!!

**Comment by pfeerick** [comment]
You're welcome! :) Glad you like it ;) And thanks for the suggestion Phil ;)

Just tested this on hardware now, RM Pocket, and the position adjustment for both columns appears to be working perfectly... effectily keeps at least a two digit spacing between the columns at all times, regardless of which one gets longer... and only moves them as necessary. Is good from 0 though the maximum ratio value of 3000.0
